analysis_driver/quality_control/bcl_validation.py

```diff
@@ -117,7 +117,7 @@ def run_bcl_check(self, bcls, slice_size=100, max_job_number=500):
             if os.path.isfile(validation_log_tmp):
                 for bcl, exit_status in self.read_check_bcl_files(validation_log_tmp):
                     f.write('%s,%s\n' % (bcl, exit_status))
-        self.info('Finished validation. Found %s invalid files' % len(self.read_invalid_files()))
+        self.info('Finished validation. Found %s invalid files', len(self.read_invalid_files()))
```

Comment:
This is the recommended way of doing string formatting in log messages. The way `logging` handles this, the formatting is only performed if the message is actually logged, whereas with `%` it's always done. I guess it's intended to avoid wasting CPU time formatting thousands of debug messages when you only have logging set to `INFO` or above (for example).

Reply:
Oh I see, so it's not calculated unless necessary.